packages/hash/task-executor/src/airbyte/executor.ts

```diff
@@ -0,0 +1,167 @@
+    } else {
+      throw new Error("Message didn't contain a ConnectorSpecification");
+    }
+  }
+
+  async runCheck(config: any): Promise<AirbyteConnectionStatus> {
+    const configPath = await writeToTempFile(
+      CONFIG_FILE_NAME,
+      JSON.stringify(config),
+    );
+
+    const response = await executeTask("docker", [
+      "run",
+      "--rm",
+      "-v",
+      `${configPath}:/secrets/config.json`,
+      this.imageName,
+      "check",
+      "--config",
+      "/secrets/config.json",
+    ]);
+
+    const messages = parseMessageStream(response);
+
+    const statusMessage = messages.find(
+      (message) =>
+        message.type === "CONNECTION_STATUS" && message.connectionStatus,
+    );
+    if (statusMessage) {
+      return statusMessage.connectionStatus!;
+    } else {
+      throw new Error("Message didn't contain a AirbyteConnectionStatus");
+    }
+  }
+
+  async runDiscover(config: any): Promise<AirbyteStream[]> {
+    const configPath = await writeToTempFile(
+      CONFIG_FILE_NAME,
+      JSON.stringify(config),
+    );
+
+    const response = await executeTask("docker", [
+      "run",
+      "--rm",
+      "-v",
+      `${configPath}:/secrets/config.json`,
+      this.imageName,
+      "discover",
+      "--config",
+      "/secrets/config.json",
+    ]);
+
+    const messages = parseMessageStream(response);
+
+    const catalogMessage = messages.find(
+      (message) => message.type === "CATALOG" && message.catalog,
+    );
+    if (catalogMessage) {
+      return catalogMessage.catalog!.streams;
+    } else {
+      throw new Error("Message didn't contain a Airbyte Catalog");
+    }
+  }
+
+  async runRead(
+    config: any,
+    configuredCatalog: ConfiguredAirbyteCatalog,
+    statePath?: string,
+  ): Promise<RecordMessage[]> {
+    const configPath = await writeToTempFile(
+      CONFIG_FILE_NAME,
+      JSON.stringify(config),
+    );
+    const catalogPath = await writeToTempFile(
+      CATALOG_FILE_NAME,
+      JSON.stringify(configuredCatalog),
+    );
+
+    const args = [
+      "run",
+      "--rm",
+      "-v",
+      `${configPath}:/secrets/config.json`,
+      "-v",
+      `${catalogPath}:/secrets/catalog.json`,
+      this.imageName,
+      "read",
+      "--config",
+      "/secrets/config.json",
+      "--catalog",
+      "/secrets/catalog.json",
+    ];
+
+    if (statePath) {
+      args.push("--state", statePath);
+    }
+
+    const response = await executeTask("docker", args);
+
+    const messages = parseMessageStream(response);
+    console.log(`Read ${messages.length} messages from Airbyte`);
+
+    const recordMessages = messages
+      .filter((message) => message.type === "RECORD" && message.record)
+      .map((message) => message.record!);
+    if (recordMessages) {
+      console.log(`There were ${recordMessages.length} records`);
+      return recordMessages;
+    } else {
+      throw new Error("Message didn't contain any Airbyte Records");
+    }
```

Comment:
**Blocking:** this is never going to throw because an empty array `[]` is truthy. You probably want to do this instead:
```suggestion
    if (recordMessages.length > 0) {
      console.log(`There were ${recordMessages.length} records`);
      return recordMessages;
    } else {
      throw new Error("Message didn't contain any Airbyte Records");
    }
```

Reply:
Good catch, I think my original error was in a try catch and wasn't about the length. (I was considering 0 records as being an acceptable state)